content/languages/cpp/igloo/stringizers.md

```diff
@@ -8,3 +8,438 @@ tags:
+  {
+    static std::string ToString(const Point3d& a)
+    {
+      std::stringstream stream;
+      stream << '(' << a.x << ", " << a.y << ", " << a.z << ')';
+      return stream.str();
+    }
+  };
+
+  //Specialization for a type you cannot modify
+  template<>
+  struct Stringizer<std::pair<int, std::string>>
+  {
+    static std::string ToString(const std::pair<int, std::string>& a)
+    {
+      std::stringstream stream;
+      stream << "(num: " << a.first << ", atr= \"" << a.second << "\")";
+      return stream.str();
+    }
+  };
+}
+
+
+//Tests
+
+It(should_pretty_print_vector_of_elements_of_non_custom_type)
+{
+  std::vector<std::pair<int, std::string>> actual   = { {1, "a"}, {2, "b"}  };
+  std::vector<std::pair<int, std::string>> expected = { {1, "a"}  };      
+  Assert::That(expected, Equals(actual));
+}
+
+It(should_pretty_print_custom_class_with_stringizer)
+{
+  Point3d start {0,  0, 0};
+  Point3d end   {7, -2, 5};
+  Assert::That(start, Equals(end));
+}
+
+It(should_pretty_print_vector_of_elements_of_custom_type_with_stringizer)
+{
+  std::vector<Point3d> actual   = { {0, 0, 0}, {2, 2, 5}  };
+  std::vector<Point3d> expected = { {1, 1, 1}  };      
+  Assert::That(expected, Equals(actual));
+}
+```
+
+This way you get meaningful output for types which were unsupported:
+
+```
+  should_pretty_print_vector_of_elements_of_non_custom_type
+    Expected: equal to [ (num: 1, atr= "a"), (num: 2, atr= "b") ]
+    Actual: [ (num: 1, atr= "a") ]
+  should_pretty_print_custom_class_with_stringizer
+    Expected: equal to (7, -2, 5)
+    Actual: (0, 0, 0)
+  should_pretty_print_vector_of_elements_of_custom_type_with_stringizer
+    Expected: equal to [ (0, 0, 0), (2, 2, 5) ]
+    Actual: [ (1, 1, 1) ]
+```
+
+This way you can also create a universal template which would solve many issues with Codewars kata: a stringizer for any pair!
+
+```cpp
+//Preloaded
+namespace snowhouse
+{
+  template<typename F, typename S>
+  struct Stringizer<std::pair<F, S>>
+  {
+    static std::string ToString(const std::pair<F, S>& a)
+    {
+      std::stringstream stream;
+      stream << '(' << a.first << ", " << a.second << ')';
+      return stream.str();
+    }
+  };
+}
+
+//Tests
+It(should_pretty_print_vector_of_pairs_of_custom_type)
+{
+  std::vector<std::pair<std::string, Point2d>> actual   = { {"A", {0, 0}}, { "B", {2, 2}}  };
+  std::vector<std::pair<std::string, Point2d>> expected = { {"A", {1, 1}}  };      
+  Assert::That(expected, Equals(actual));
+}
+
+```
+
+You can get meaningful assertion messages for any `std::pair<F, T>`, as long as types `F` and `T` can also be stringified, what is usually the case for Codewars kata:
+
+```
+should_pretty_print_vector_of_pairs_of_custom_type
+  Expected: equal to [ (A, (0, 0)), (B, (2, 2)) ]
+  Actual: [ (A, (1, 1)) ]
+```
+
+## Additional info
+ - [Getting better output for your types](https://github.com/banditcpp/snowhouse#getting-better-output-for-your-types) in Snowhouse repository.
+
```

Comment:
Add link to kumite?

https://www.codewars.com/kumite/5f9b749a9cd7a8002dfa2f34/

Reply:
```suggestion
 - [Kumite](https://www.codewars.com/kumite/5f9b749a9cd7a8002dfa2f34?sel=5f9b749a9cd7a8002dfa2f34) which illustrates how to use stringification in a kata.
```